src/macaron/repo_finder/commit_finder.py

```diff
@@ -0,0 +1,397 @@
+                this_version_pattern = this_version_pattern + "(" + INFIX_2 + "0)?"
+                numeric_parts = numeric_parts + 1
+
+    this_version_pattern = f"{PREFIX}{PREFIX_SEPARATOR}(?P<version>{this_version_pattern}){SUFFIX_SEPARATOR}{SUFFIX}$"
+    return re.compile(this_version_pattern, flags=re.IGNORECASE), parts, has_non_numeric_suffix
+
+
+def _match_tags(
+    tag_list: list[TagReference], pattern: Pattern, parts: list[str], has_non_numeric_suffix: bool, artifact_name: str
+) -> list[TagReference]:
+    """Return items of the passed tag list that match the passed pattern.
+
+    Parameters
+    ----------
+    tag_list: list[TagReference]
+        The list of tags to check.
+    pattern: Pattern
+        The pattern to match against.
+    parts: list[str]
+        The list of version parts extracted from the version.
+    has_non_numeric_suffix : bool
+        Whether the version string has a non-numeric suffix.
+    artifact_name : str
+        The name of the artifact.
+
+    Returns
+    -------
+    list[TagReference]
+        The list of tags that matched the pattern.
+    """
+    matched_tags = []
+    for tag in tag_list:
+        tag_name = str(tag)
+        match = pattern.match(tag_name)
+        if not match:
+            continue
+        # Tags are append with their match information for possible further evaluation.
+        matched_tags.append(
+            {
+                "tag": tag,
+                "version": match.group("version"),
+                "prefix": match.group("prefix"),
+                "prefix_sep": match.group("prefix_sep"),
+                "suffix_sep": match.group("suffix_sep"),
+                "suffix": match.group("suffix"),
+            }
+        )
+
+    if len(matched_tags) <= 1:
+        return _extract_tags_from_list(matched_tags)
+
+    # In the case of multiple matches, further work must be done.
+    # Firstly, combine matches with their suffixes as some version patterns will not include the required suffix in the
+    #  version group.
+    if has_non_numeric_suffix:
+        filtered_tags = []
+        for item in matched_tags:
+            # Discard tags with no suffix or with one that does not match the version.
+            suffix: str | None = item["suffix"]
+            if not suffix:
+                filtered_tags.append(item)
+                continue
+            if suffix == parts[len(parts) - 1]:
+                filtered_tags.append(item)
+                continue
+
+        matched_tags = filtered_tags
+
+    # If any of the matches contain a prefix that matches the target artifact name, remove those that don't.
+    named_tags = []
+    for item in matched_tags:
+        prefix: str | None = item["prefix"]
+        if not prefix:
+            continue
+        if "/" in prefix:
+            # Exclude prefix parts that exists before a forward slash, e.g. rel/
+            split = prefix.split("/")
+            prefix = split[len(split) - 1]
```

Comment:
Suggestion: this is probably cleaner:
```suggestion
        # Exclude prefix parts that exist before a forward slash, e.g. rel/
        _, _, prefix = prefix.rpartition("/")
```